src/platform/user/profile/vap-svc/util/getProfileInfoFieldAttributes.js

```diff
@@ -71,6 +72,8 @@ export const getProfileInfoFieldAttributes = fieldName => {
 
     if (fieldName === FIELD_NAMES.MOBILE_PHONE) {
       title = FIELD_TITLES[FIELD_NAMES.MOBILE_PHONE];
+      delete uiSchema.extension; // Remove for mobile only
```

Comment:
nit: not critical, but you can avoid using `cloneDeep` here by using `omit` instead, which might be a tad quicker than the up-front cloning:
```
uiSchema = omit(uiSchema, 'extension');
formSchema = omit(formSchema, 'properties.extension');
```